### ISSUE TITLE
Installation error in Windows

### ISSUE THREAD

**Issue by gperversi** [tproffen/DiffuseCode#11]
Good morning, 
I'm trying to install discus on my work computer (windows10) but even following the procedure of the pdf installation document to a T every time I run into the same error, which seems to be that the program doesn't create all the directories that it will later call, and doesn't actually provide a bat file.

Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d-----        28-6-2021     10:29                DISCUS_INSTALLATION
Ubuntu2004 is already installed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   128  100   128    0     0    128      0  0:00:01 --:--:--  0:00:01   390
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   625  100   625    0     0    625      0  0:00:01 --:--:--  0:00:01  1213
100  4428  100  4428    0     0   4428      0  0:00:01 --:--:--  0:00:01  4428
Access is denied.
 COPY DiscusWsl
Copy-item : Cannot find path 'C:\Users\DISCUS_INSTALLATION\DiscusWSL\' because it does not exist.
At C:\Users\P70067115\Downloads\bbb_install_suite_Windows10_WSL.ps1:96 char:1
+ Copy-item -Recurse -Force DiscusWSL\ -Destination "C:\Program Files ( ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\DISCUS_INSTALLATION\DiscusWSL\:String) [Copy-Item], ItemNotFou
   ndException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand

W_USER              P70067115
SourceFileLocation  C:\Program Files (x86)\DiscusWSL\discus_suite_ps1.bat
ShortcutLocation    C:\Users\P70067115\Desktop\DiscusSUITE_WSL.lnk
VcXsrv is already installed
& : The term 'C:\Program Files (x86)\DiscusWSL\discus_suite_ps1.bat' is not recognized as the name of a cmdlet,
function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the
path is correct and try again.
At C:\Users\P70067115\Downloads\bbb_install_suite_Windows10_WSL.ps1:131 char:3
+ & "C:\Program Files (x86)\DiscusWSL\discus_suite_ps1.bat"
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Program File...s_suite_ps1.bat:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Could you advise on how to fix this?

Cheers,
Giuditta

**Comment by rneder** [comment]
Hi Giuditta,
I receive your error message if I start the powershell as regular user, not in administrator mode. 
Please make sure the powershell is started with administrator rights, as the installation script needs to create/write folders like:
C:\Users\DISCUS_INSTALLATION\ 
and C:\Programs (x86)\DiscusWSL
Is the home folder of the user account that you start the powershell at C:\Users or is it at a different disk?
Best
Reinhard

**Comment by gperversi** [comment]
Hi Reinhard,

Thank you for the prompt reply! I had a previous version of discus installed but trying to put the new one is proving challenging...


I'm running it as administrator already, but it's giving the error regardless.

The home folder is C:\Users\P70067115. Should I edit the .ps1 file accordingly?

I can see that at the moment the DISCUS_INSTALLATION file gets created outside of the C:\Users\P70067115 (just in the previous layer, Users).



Cheers,

Giuditta




Dr Giuditta Perversi
Lecturer in Chemistry
Maastricht Science Programme (MSP)
***@***.*** ***@***.***>
www.maastrichtuniversity.nl <http://www.maastrichtuniversity.nl/>

Paul Henri Spaaklaan 1, 6229 EN Maastricht, The Netherlands
Office Room #B2.010
P.O. Box 616, 6200 MD Maastricht, The Netherlands
T +31 43 38 85193

Disclaimer<http://www.maastrichtuniversity.nl/web/show/id=6568779/langid=42>
[X] Please consider your environmental responsibility before printing this e-mail.
________________________________

**Comment by black-06** [comment]
Hi Reinhard,
The latest version on windows has some bugs for installation.
I find the powershell script download the bbb_install_script.sh in C:\\Users\\DISCUS_INSTALLTION,
and the base shell script want to download something from github(like DIFFUSE_INSTALL.tar.gz/DIFFUSE_CODE_pgplot.tar.gz)
but it is not working in  line 85:
`export DISCUS_VERSION=$(curl --silent "https://github.com/tproffen/DiffuseCode/releases/latest" | grep -Poe 'v.[0-9]*.[0-9]*.[0-9]*')`
It can not get the right version ! it return NULL !!!

It causes the correct link ("https://github.com/rneder/DiffuseSuplement/releases/V.1.0.0/DIFFUSE_INSTALL.tar.gz")to become wrong, such as "https://github.com/rneder/DiffuseSuplement/releases/download//DIFFUSE_INSTALL.tar.gz"

The same error occurs at line 88:
`export DISCUS_SUPPLEMENT=$(curl --silent "https://github.com/rneder/DiffuseSuplement/releases/latest" | grep -Poe 'v.[0-9]*.[0-9]*.[0-9]*')`
at Ubuntu(WSL) /home/xx_user_name/DIFFUSE_INSTALL/get_diffuse_ubuntu.sh line 9:
`export DISCUS_SUPPLEMENT=$(curl --silent "https://github.com/rneder/DiffuseSuplement/releases/latest" | grep -Poe 'v.[0-9]*.[0-9]*.[0-9]*')`

In summary, these files cannot be downloaded correctly, causing the installation to fail on windows.
And I would like to contribute to DiffuseCode when i have time.
(And I am not a English native speaker,please forgive me)

**Comment by rneder** [comment]
Hi "black-06"
I cannot reproduce the error message. 
If I download the latest Windows10 installation script bbb_install_suite_Windows10_WSL.ps1 and execute this from within a powershell that I have started with administrator rights, everything installs just fine. This has been tested on:
Windows10 with and without WSL/Ubuntu installed. 
Both on systems with WSL 1 and WSL 2
It does install 6.08.01 

Could you please try the installation again  with the latest bbb_install_suite_Windows10_WSL.ps1 from the current release (v.6.08.01)?
Could you please document the installation issue in a bit more detail:
Did you download the latest bbb_install_suite_Windows10_WSL.ps1 ?
Was a previous version of discus_suite installed on your computer, if so which version?
As default, I use WSL1 instead of WSL 2, Which version is running on your computer? To find out, please start the powershell and run: wsl -l -v
could you send screen shots of the error message ?

**Comment by drtaoq** [comment]
> Hi  Giuditta，I met excatly the same problem with yours. I wonder if you had well soleved it and how?  Thank you! Cheers Qi



> Hi Reinhard, Thank you for the prompt reply! I had a previous version of discus installed but trying to put the new one is proving challenging... I'm running it as administrator already, but it's giving the error regardless. The home folder is C:\Users\P70067115. Should I edit the .ps1 file accordingly? I can see that at the moment the DISCUS_INSTALLATION file gets created outside of the C:\Users\P70067115 (just in the previous layer, Users). Cheers, Giuditta

**Comment by rneder** [comment]
Hi Qi,
The error that Giuditta encountered was caused by the antivirus software, which did not allow the Windows Subsystem for Linux(WSL) to be installed properly. With the thread prevention setting to allow WSL the installation went smoothly. 
Another issue I have encountered is related to WSL version 2. With WSL version 2 one might have to tweak the settings for virtual machines on some Windows installation. Thus, please ensure that after the initial installation of WSL with the dism.exe program:
dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart

you issue the command in a powershell window:

wsl --set-default-version 1
 
An error I hav encountered is  users not to enter the dism.exe command in a single command line.

**Comment by gperversi** [comment]
Hi Qi!
I confirm Reinhard response above, basically my work computer has McAfee and it blocks the Linux environment to execute it properly. If you have the same antivirus, you can go into Settings (from the tiny arrow up right in the window) > Threat Prevention > Show advanced. The thing you're looking for is in "Rules" under the name "Executing Windows Subsystem for Linux". Untick the two boxed and you should be able to proceed with the installation, but mind that this setting is stubborn and might come back by itself!

Cheers,
Giuditta

**Comment by drtaoq** [comment]
Hi Reinhard and Giuditta，

Thank you for your replies. I have a good news and a bad news to update. Good news is that I have successfully installed DISCUS using an Ubuntu with VM. Bad news is that I cannot still finish the install process using Ubuntu-LTS.  I have no McAfee or any other antiviral software in my PC, so I do not know. It was likely caused by the instability network of mine to visit GitHub. 

When I use the scripts (wget https://github.com/tproffen/DiffuseCode/releases/download/v.6.08.02/bbb_install_script.sh && chmod u+x * && ./bbb_install_script.sh) under Ubuntu-LTS, it do can download installation parts and fetch thousands of Ubuntu related archives. However, it was stuck every time at the near end, when it starts to download files from github, with a message like (DISCUS_SUITE INSTALLATION
https://github.com/rneder/DiffuseSuplement/releases/download//DIFFUSE_WSL_Ubuntu_2004_v.6.08.02.tar.gz does not exist
 Installation of DISCUS_SUITE from code at github
 Download may take a moment, please be patient
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:21 --:--:--     0
curl: (7) Failed to connect to github.com port 443: Connection refused).

I would try some more times at different times of a day, and see what it will happen. Enjoy your holiday! Cheers Qi.

**Comment by drtaoq** [comment]
Hi Reinhard and Giuditta，

Thank you for your replies. I have a good news and a bad news to update. Good news is that I have successfully installed DISCUS using an Ubuntu with VM. Bad news is that I cannot still finish the install process using Ubuntu-LTS.  I have no McAfee or any other antiviral software in my PC, so I do not know. It was likely caused by the instability network of mine to visit GitHub. 

When I use the scripts (wget https://github.com/tproffen/DiffuseCode/releases/download/v.6.08.02/bbb_install_script.sh && chmod u+x * && ./bbb_install_script.sh) under Ubuntu-LTS, it do can download installation parts and fetch thousands of Ubuntu related archives. However, it was stuck every time at the near end, when it starts to download files from github, with a message like (DISCUS_SUITE INSTALLATION
https://github.com/rneder/DiffuseSuplement/releases/download//DIFFUSE_WSL_Ubuntu_2004_v.6.08.02.tar.gz does not exist
 Installation of DISCUS_SUITE from code at github
 Download may take a moment, please be patient
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:21 --:--:--     0
curl: (7) Failed to connect to github.com port 443: Connection refused).

I would try some more times at different times of a day, and see what it will happen. Enjoy your holiday! Cheers Qi.

**Comment by rneder** [comment]
It appears that you seem to be behind a proxy server / firewall that does not seem to grant access to github?
Within powershell please try:
curl.exe "https://github.com/rneder/DiffuseSuplement/releases/latest"
Alternatively try within the WSL Ubuntu the command
curl "https://github.com/rneder/DiffuseSuplement/releases/latest"

In both cases you should get a response:
<html><body>You are being <a href="https://github.com/rneder/DiffuseSuplement/releases/tag/v.1.0.0">redirected</a>.</body></html>
What happens?

**Comment by drtaoq** [comment]
> Thank you, Reinhard.
Both reply that "<html><body>You are being <a href="https://github.com/rneder/DiffuseSuplement/releases/tag/v.1.0.0">redirected</a>.</body></html>".
I am not sure about proxy or redirection parts.

**Comment by drtaoq** [comment]
I guess I know what has happened. Because my network is not stable, I add an IP adress of github.com to my hosts file.  When my network was relatively smooth, I performed the installation again and finally got the discus start.

![image](https://user-images.githubusercontent.com/95006901/143581450-93d47aa9-532f-486d-8f0a-65ef012dec1a.png)

**Comment by rneder** [comment]
Glad to hear that the installation worked out!
